migration-engine/migration-engine-tests/tests/migrations/indexes.rs

```diff
@@ -406,7 +406,7 @@ async fn index_updates_with_rename_must_work(api: &TestApi) -> TestResult {
         .table_bang("A")
         .indices
         .iter()
-        .filter(|i| i.columns == &["field", "id"] && i.name == "customNameA");
+        .filter(|i| i.columns == ["field", "id"] && i.name == "customNameA");
```

Comment:
what is the rationale for these diffs?

Reply:
Clippy warned about needless borrows; the right-hand borrow is not required here. This is part of the [op_ref](https://rust-lang.github.io/rust-clippy/master/index.html#op_ref) lint.

Reply:
It usually would suggest converting this into `*i.columns == ["field", "id"]`, but since no borrow is required at all it suggests to not borrow instead.

Reply:
I guess the rationale is that rather than adding pointers so types align, it's preferable to dereference pointers instead. Not the most impactful here, but as a general rule it's not one I disagree with.